auth-api/src/auth_api/services/user.py

```diff
@@ -81,69 +83,153 @@ def create_user_and_add_membership(memberships: List[dict], org_id, token_info:
         single_mode= true is used now incase of invitation for admin users scenarion
         other cases should be invoked with single_mode=false
         """
-        if single_mode:  # make sure no bulk operation and only owner is created using if no auth
-            if len(memberships) > 1 or memberships[0].get('membershipType') not in [OWNER, ADMIN]:
-                raise BusinessException(Error.INVALID_USER_CREDENTIALS, None)
-        else:
-            check_auth(org_id=org_id, token_info=token_info, one_of_roles=(ADMIN, OWNER))
-
-        # check if anonymous org ;these actions cannot be performed on normal orgs
-        org = OrgModel.find_by_org_id(org_id)
-        if not org or org.access_type != AccessType.ANONYMOUS.value:
-            raise BusinessException(Error.INVALID_INPUT, None)
+        User._validate_and_throw_exception(memberships, org_id, single_mode, token_info)
 
         current_app.logger.debug('create_user')
         users = []
         for membership in memberships:
-
-            current_app.logger.debug(f"create user username: {membership['username']}")
+            username = membership['username']
+            current_app.logger.debug(f'create user username: {username}')
             create_user_request = User._create_kc_user(membership)
-            db_username = IdpHint.BCROS.value + '/' + membership['username']
-            existing_user = UserModel.find_by_username(db_username)
-            if existing_user:
+            db_username = IdpHint.BCROS.value + '/' + username
+            user_model = UserModel.find_by_username(db_username)
+            re_enable_user = False
+            existing_kc_user = KeycloakService.get_user_by_username(username)
+            enabled_in_kc = getattr(existing_kc_user, 'enabled', True)
+            if getattr(user_model, 'status', None) == Status.INACTIVE.value and not enabled_in_kc:
```

Comment:
Its better to stick with .get(key, default value) instead of getattr for consistency.

Reply:
ok..Let me check why I went for it.